AmplifyPlugins/Core/AWSPluginsCore/Model/Decorator/ModelIdDecorator.swift

```diff
@@ -27,10 +39,23 @@ public struct ModelIdDecorator: ModelBasedGraphQLDocumentDecorator {
         var inputs = document.inputs
 
         if case .mutation = document.operationType {
+            var objectMap = [String: String]()
+            if let fields = fields {
+                for (fieldName, value) in fields {
+                    objectMap[fieldName] = value
+                }
+            }
+            objectMap["id"] = id
             inputs["input"] = GraphQLDocumentInput(type: "\(document.name.pascalCased())Input!",
-            value: .object(["id": id]))
+                                                   value: .object(objectMap))
         } else if case .query = document.operationType {
             inputs["id"] = GraphQLDocumentInput(type: "ID!", value: .scalar(id))
+
+            if let fields = fields {
+                for (fieldName, value) in fields {
+                    inputs[fieldName] = GraphQLDocumentInput(type: "String!", value: .scalar(value))
+                }
+            }
```

Comment:
if `id` is included in the provided `fields`, are we overriding its value in `inputs` set on line 52 but with a different type? (`String!` instead of `ID!`)

Reply:
good point, that's what would happen in the current code. passing in `fields: ["id": "123"]` will have unintended behavior. Since this is a public facing type, we should make sure the constructor can be used in different ways without causing problems. I'm not sure what's a better interface that also keeps it backwards compatible. perhaps the best we can do right now is to make it work for the customer even when they pass in parameters in diferent ways

although this can be passed in, what is the expected behavior? `init(id: "111", fields: ["id": "222"])`` do they want the id to be "111" or "222". We should simply ignore the "id" keys in fields if it exists, and add doc comment to the initializer to provide an example of usage, indicating that fields is to contain additional index values

Reply:
Added changes to ignore `id` in `fields` when decoration is being performed.